File: monkey.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 12 11:23:45 2015

@author: Marcus Therkildsen
"""
from __future__ import division
import itertools

'''
Reason for making this script: https://www.youtube.com/watch?v=U9qU20VmvaU

5 sailors, a monkey and a giant pile of coconuts.
Each sailor wakes up one at a time.
The awoken sailor sorts the coconuts into 5 piles.
There is 1 coconut left over -> the monkey gets it.
Then the sailor takes his pile and stores it somewhere.
'''

sailors = 5
monkeys = 1

for coconuts_tot in itertools.count():
    coconuts = coconuts_tot
    for sailor in xrange(sailors):
        # One for each monkey
        coconuts -= monkeys
        # Divide equally among sailors
        if coconuts % sailors:
            break
        coconuts -= coconuts//sailors
    if not coconuts % sailors:
        break

print('Solution: ' + str(coconuts_tot) + ' coconuts to begin with')

File: monkey.cs
using System;
/**
 * Created by Marcus Therkildsen on 27-10-2015.
 * Using http://www.tutorialspoint.com/compile_csharp_online.php 
 * to check if working
 */
public class MonkeyCSharp {
    
    // This is the main, equivalent to if __name__ == '__main__': in python
    public static void Main(String[] args) {
        
    // Creating some start variables
    int sailors = 5;
    int monkeys = 1;
    int j;
    int coconuts_tot = 0;
    int coconuts;
	
    // Go through the number of coconuts
    while (0 == 0){
	coconuts_tot += 1;
        coconuts = coconuts_tot;
        
        // Go through the number of sailors
        for (j=0; j<sailors;j++){

            // One for each monkey
            coconuts -= monkeys;      
            if (coconuts % sailors != 0){
                break;
			}			
            coconuts -= coconuts/sailors;	
            }
		
        if (coconuts % sailors == 0){
	    break;        
	    } 
    }
    Console.Write("Solution: " + coconuts_tot + " coconuts to begin with");		
    }
}
